### DISCUSSION TITLE
Is there an Unreliable Ordered or Sequenced option?

### DISCUSSION THREAD

**Discussion by cakeslice** [FirstGearGames/FishNet#244] [Q&A - DO NOT USE - ASK ON DISCORD]
Getting a packet out of order is always a bad thing and an easy solution is to drop packets that are older than the ones already processed.

Is there any support for this in FishNet or the transports it uses? Or is this already done by default?

Also here's an interesting post about this for the game FortNite: https://www.reddit.com/r/FortNiteBR/comments/awagpo/packet_reordering_technical_post/

**Comment by FirstGearGames** [comment]
There is no option for this in the default transport. Its either reliable ordered, or unreliable unordered. FishNet already takes into account unordered packets.

**Reply by cakeslice** [reply]
I've been taking a look at the code and it's nice that there's packetId checks that discard old data.

However, there is still the question of sorting packets before processing them.

Example:
    - Client sends two RPCs `Shoot()` and `Reload()`, in that order
    - Server gets both RPCs in different packets and out of order. Will apply first `Reload()` and then `Shoot()`
 
The problem is that in the server the player won't be able to shoot because it's now reloading while the client though he was able to.

Does FishNet sort all packets in the buffer before processing them or always does it by received order?
If there is no sorting, is it because it could complicate things due to packet splitting?

**Reply by FirstGearGames** [reply]
If you are using the reliable channel they will always arrive in order. If using unreliable there is no way to know for sure which order they arrive, if they arrive at all.
